libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/RouteUtils.java

```diff
@@ -136,6 +138,25 @@ public boolean isLastLeg(RouteProgress routeProgress) {
     return coordinates;
   }
 
+  /**
+   * Given a {@link RouteProgress}, this method will calculate the remaining waypoint names
+   * along the given route based on route option waypoint names and the progress remaining coordinates.
+   *
+   * @param routeProgress for route waypoint names and remaining coordinates
+   * @return String array of remaining waypoint names
+   * @since 0.19.0
+   */
+  @Nullable
+  public String[] calculateRemainingWaypointNames(RouteProgress routeProgress) {
```

Comment:
If this is only used in `RouteFetcher#addWaypointNames` why we need to expose it as a `public` API here in `RouteUtils`?

IMO, objects and methods should live/be where their context is. From my experience utility classes become a drawer easily and we should treat them carefully.